src/main/java/org/prebid/server/settings/CompositeApplicationSettings.java

```diff
@@ -0,0 +1,144 @@
+    public Future<StoredRequestResult> getStoredRequestsByAmpId(Set<String> ids, Timeout timeout) {
+        return proxy.getStoredRequestsByAmpId(ids, timeout);
+    }
+
+    private static class Proxy implements ApplicationSettings {
+
+        private ApplicationSettings applicationSettings;
+        private Proxy next;
+
+        Proxy(ApplicationSettings applicationSettings, Proxy next) {
+            this.applicationSettings = applicationSettings;
+            this.next = next;
+        }
+
+        @Override
+        public Future<Account> getAccountById(String accountId, Timeout timeout) {
+            final BiFunction<String, Timeout, Future<Account>> nextRetriever =
+                    next != null ? next::getAccountById : null;
+            return getConfig(accountId, timeout, applicationSettings::getAccountById, nextRetriever);
+        }
+
+        @Override
+        public Future<String> getAdUnitConfigById(String adUnitConfigId, Timeout timeout) {
+            final BiFunction<String, Timeout, Future<String>> nextRetriever =
+                    next != null ? next::getAdUnitConfigById : null;
+            return getConfig(adUnitConfigId, timeout, applicationSettings::getAdUnitConfigById, nextRetriever);
+        }
+
+        private static <T> Future<T> getConfig(String key, Timeout timeout,
+                                               BiFunction<String, Timeout, Future<T>> retriever,
+                                               BiFunction<String, Timeout, Future<T>> nextRetriever) {
+            return retriever.apply(key, timeout)
+                    .recover(throwable -> nextRetriever != null
+                            ? nextRetriever.apply(key, timeout)
+                            : Future.failedFuture(throwable));
+        }
+
+        @Override
+        public Future<StoredRequestResult> getStoredRequestsById(Set<String> ids, Timeout timeout) {
+            final BiFunction<Set<String>, Timeout, Future<StoredRequestResult>> nextRetriever =
+                    next != null ? next::getStoredRequestsById : null;
+            return getStoredRequests(ids, timeout, applicationSettings::getStoredRequestsById, nextRetriever);
+        }
+
+        @Override
+        public Future<StoredRequestResult> getStoredRequestsByAmpId(Set<String> ids, Timeout timeout) {
+            final BiFunction<Set<String>, Timeout, Future<StoredRequestResult>> nextRetriever =
+                    next != null ? next::getStoredRequestsByAmpId : null;
+            return getStoredRequests(ids, timeout, applicationSettings::getStoredRequestsByAmpId, nextRetriever);
+        }
+
+        private static Future<StoredRequestResult> getStoredRequests(
+                Set<String> ids, Timeout timeout,
+                BiFunction<Set<String>, Timeout, Future<StoredRequestResult>> retriever,
+                BiFunction<Set<String>, Timeout, Future<StoredRequestResult>> nextRetriever) {
+
+            return retriever.apply(ids, timeout)
```

Comment:
If any backend fails, even the first one, the whole request fails as well. Is it intended behavior? I was under impression that until there are available backends we may query them since they may have a valid answer.

Reply:
For stored request fetching we don't expect throwing exceptions. Instead any errors occured while processing will be present in `org.prebid.server.settings.model.StoredRequestResult#errors` list.
So, the whole chain will be processed.